pkg/node/manager/manager.go

```diff
@@ -362,7 +362,7 @@ func (m *Manager) NodeUpdated(n nodeTypes.Node) {
 		// If the host firewall is enabled, all traffic to remote nodes must go
 		// through the tunnel to preserve the source identity as part of the
 		// encapsulation.
-		if address.Type == addressing.NodeCiliumInternalIP || m.conf.NodeEncryptionEnabled() ||
+		if address.Type == addressing.NodeCiliumInternalIP || m.conf.EncryptionEnabled() ||
```

Comment:
We're accruing a lot of conditionals here, I'm wondering what the inverse condition now looks like? Is this getting towards `if tunnelModeEnabled { ... }`?

Reply:
Looks like this is the same question that @jrfastab asks here ;)
https://github.com/cilium/cilium/pull/14599#issuecomment-759775002

Reply:
Yep overall I think we need to start reducing the 'if encrypt { foo }' cases to simplify things.